Cannot compile Fettle from source using MSBuild 15.8 (Visual Studio 2017 version 15.8.0)

AppVeyor switched the "Visual Studio 2017" profile from MSBuild version 15.7 to 15.8. As soon as this happened, Fettle's CI started producing compiler warnings and test failures, failing the build.

For now, Fettle's CI settings have been modified to use the previous version and builds are green again, but this should be looked into.

Compile warnings:
```
Microsoft (R) Build Engine version 15.8.166+gd4e8d81a88 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3277: Found conflicts between different versions of "System.Collections.Immutable" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\projects\fettle\src\Core\Core.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3277: Found conflicts between different versions of "System.Collections.Immutable" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\projects\fettle\src\Console\Console.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "dotMemory.Unit, Version=109.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [C:\projects\fettle\src\Tests\Tests.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3277: Found conflicts between different versions of "System.Collections.Immutable" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\projects\fettle\src\Tests\Tests.csproj]
```

Test failures:
```
Fettle.Tests.Console.EndToEnd.Happy_path.Console_executable_runs_and_outputs_info_about_surviving_mutants
  Expected: 1
  But was:  3
   at Fettle.Tests.Console.EndToEnd.Happy_path.<>c__DisplayClass0_0.<Console_executable_runs_and_outputs_info_about_surviving_mutants>b__0() in C:\projects\fettle\src\Tests\Console\EndToEnd\Happy_path.cs:line 45
   at NUnit.Framework.Assert.Multiple(TestDelegate testDelegate) in C:\src\nunit\nunit\src\NUnitFramework\framework\Assert.cs:line 333
   at Fettle.Tests.Console.EndToEnd.Happy_path.Console_executable_runs_and_outputs_info_about_surviving_mutants() in C:\projects\fettle\src\Tests\Console\EndToEnd\Happy_path.cs:line 49
1-2) Failed : Fettle.Tests.Console.EndToEnd.Happy_path.Console_executable_runs_and_outputs_info_about_surviving_mutants
  Expected: String containing "mutant(s) survived!"
  But was:  "Analysing test coverageAn error ocurred that Fettle didn't expect.
System.AggregateException: One or more errors occurred. ---> System.IO.FileLoadException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(IDictionary`2 properties)
   at Fettle.Core.MSBuildWorkspaceFactory.Create() in C:\projects\fettle\src\Core\MSBuildWorkspaceFactory.cs:line 16
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 44
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Fettle.Console.Program.AnalyseCoverage(ICoverageAnalyser coverageAnalyser, IOutputWriter outputWriter, Config config) in C:\projects\fettle\src\Console\Program.cs:line 109
   at Fettle.Console.Program.InternalEntryPoint(String[] args, Func`3 mutationTestRunnerFactory, Func`2 coverageAnalyserFactory, IOutputWriter outputWriter) in C:\projects\fettle\src\Console\Program.cs:line 55
---> (Inner Exception #0) System.IO.FileLoadException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(IDictionary`2 properties)
   at Fettle.Core.MSBuildWorkspaceFactory.Create() in C:\projects\fettle\src\Core\MSBuildWorkspaceFactory.cs:line 16
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 44
WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
<---
"
   at Fettle.Tests.Console.EndToEnd.Happy_path.<>c__DisplayClass0_0.<Console_executable_runs_and_outputs_info_about_surviving_mutants>b__0() in C:\projects\fettle\src\Tests\Console\EndToEnd\Happy_path.cs:line 47
   at NUnit.Framework.Assert.Multiple(TestDelegate testDelegate) in C:\src\nunit\nunit\src\NUnitFramework\framework\Assert.cs:line 333
   at Fettle.Tests.Console.EndToEnd.Happy_path.Console_executable_runs_and_outputs_info_about_surviving_mutants() in C:\projects\fettle\src\Tests\Console\EndToEnd\Happy_path.cs:line 49
1-3) Failed : Fettle.Tests.Console.EndToEnd.Happy_path.Console_executable_runs_and_outputs_info_about_surviving_mutants
  Expected: String containing "PartiallyTestedNumberComparison.cs:7"
  But was:  "Analysing test coverageAn error ocurred that Fettle didn't expect.
System.AggregateException: One or more errors occurred. ---> System.IO.FileLoadException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(IDictionary`2 properties)
   at Fettle.Core.MSBuildWorkspaceFactory.Create() in C:\projects\fettle\src\Core\MSBuildWorkspaceFactory.cs:line 16
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 44
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Fettle.Console.Program.AnalyseCoverage(ICoverageAnalyser coverageAnalyser, IOutputWriter outputWriter, Config config) in C:\projects\fettle\src\Console\Program.cs:line 109
   at Fettle.Console.Program.InternalEntryPoint(String[] args, Func`3 mutationTestRunnerFactory, Func`2 coverageAnalyserFactory, IOutputWriter outputWriter) in C:\projects\fettle\src\Console\Program.cs:line 55
---> (Inner Exception #0) System.IO.FileLoadException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
File name: 'System.Collections.Immutable, Version=1.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create(IDictionary`2 properties)
   at Fettle.Core.MSBuildWorkspaceFactory.Create() in C:\projects\fettle\src\Core\MSBuildWorkspaceFactory.cs:line 16
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 44
WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
<---
"
   at Fettle.Tests.Console.EndToEnd.Happy_path.<>c__DisplayClass0_0.<Console_executable_runs_and_outputs_info_about_surviving_mutants>b__0() in C:\projects\fettle\src\Tests\Console\EndToEnd\Happy_path.cs:line 48
   at NUnit.Framework.Assert.Multiple(TestDelegate testDelegate) in C:\src\nunit\nunit\src\NUnitFramework\framework\Assert.cs:line 333
   at Fettle.Tests.Console.EndToEnd.Happy_path.Console_executable_runs_and_outputs_info_about_surviving_mutants() in C:\projects\fettle\src\Tests\Console\EndToEnd\Happy_path.cs:line 49
2) SetUp Error : Fettle.Tests.Core.Coverage.Happy_path
System.AggregateException : One or more errors occurred.
  ----> System.Exception : Failed to compile project HasSurvivingMutants.Implementation
warning CS8021: No value for RuntimeMetadataVersion found. No assembly containing System.Object was found nor was a value for RuntimeMetadataVersion specified through options.
error CS5001: Program does not contain a static 'Main' method suitable for an entry point
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Fettle.Tests.Core.Contexts.Coverage.When_analysing_coverage(Boolean catchExceptions) in C:\projects\fettle\src\Tests\Core\Contexts\Coverage.cs:line 104
--Exception
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileProject>d__7.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 0
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileMultipleProjects>d__6.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 100
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 55
3) SetUp Error : Fettle.Tests.Core.Coverage.Source_file_filters
System.AggregateException : One or more errors occurred.
  ----> System.Exception : Failed to compile project HasSurvivingMutants.Implementation
warning CS8021: No value for RuntimeMetadataVersion found. No assembly containing System.Object was found nor was a value for RuntimeMetadataVersion specified through options.
error CS5001: Program does not contain a static 'Main' method suitable for an entry point
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Fettle.Tests.Core.Contexts.Coverage.When_analysing_coverage(Boolean catchExceptions) in C:\projects\fettle\src\Tests\Core\Contexts\Coverage.cs:line 104
--Exception
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileProject>d__7.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 0
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileMultipleProjects>d__6.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 100
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 55
4) SetUp Error : Fettle.Tests.Core.Coverage.Tests_fail
System.AggregateException : One or more errors occurred.
  ----> System.Exception : Failed to compile project HasSurvivingMutants.Implementation
warning CS8021: No value for RuntimeMetadataVersion found. No assembly containing System.Object was found nor was a value for RuntimeMetadataVersion specified through options.
error CS5001: Program does not contain a static 'Main' method suitable for an entry point
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Fettle.Tests.Core.Contexts.Coverage.When_analysing_coverage(Boolean catchExceptions) in C:\projects\fettle\src\Tests\Core\Contexts\Coverage.cs:line 104
--Exception
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileProject>d__7.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 0
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileMultipleProjects>d__6.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 100
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 55
5) Error : Fettle.Tests.Core.Coverage_analysis.Then_only_tests_that_cover_members_are_run
Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: r => r.RunTests(It.IsAny<IEnumerable`1>(), It.Is<IEnumerable`1>(tn => tn.Contains<String>("example.test.one")))
Configured setups: 
x => x.RunTests(It.IsAny<IEnumerable`1>(), It.IsAny<IEnumerable`1>())
No invocations performed.
   at Moq.Mock.ThrowVerifyException(MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, Expression expression, Times times, Int32 callCount) in C:\projects\moq4\Source\Mock.cs:line 400
   at Moq.Mock.VerifyCalls(Interceptor targetInterceptor, MethodCall expected, Expression expression, Times times) in C:\projects\moq4\Source\Mock.cs:line 385
   at Moq.Mock.Verify[T,TResult](Mock`1 mock, Expression`1 expression, Times times, String failMessage) in C:\projects\moq4\Source\Mock.cs:line 321
   at Fettle.Tests.Core.Coverage_analysis.Then_only_tests_that_cover_members_are_run() in C:\projects\fettle\src\Tests\Core\Coverage_analysis.cs:line 20
6) SetUp Error : Fettle.Tests.Core.EndToEnd.Happy_path
System.AggregateException : One or more errors occurred.
  ----> System.Exception : Failed to compile project HasSurvivingMutants.Implementation
warning CS8021: No value for RuntimeMetadataVersion found. No assembly containing System.Object was found nor was a value for RuntimeMetadataVersion specified through options.
error CS5001: Program does not contain a static 'Main' method suitable for an entry point
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Fettle.Tests.Core.Contexts.EndToEnd.When_mutation_testing_the_app() in C:\projects\fettle\src\Tests\Core\Contexts\EndToEnd.cs:line 34
--Exception
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileProject>d__7.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 0
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<InstrumentThenCompileMultipleProjects>d__6.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 100
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Fettle.Core.CoverageAnalyser.<AnalyseCoverage>d__5.MoveNext() in C:\projects\fettle\src\Core\CoverageAnalyser.cs:line 55
7) Failed : Fettle.Tests.Core.Events.Then_event_listener_is_called_when_events_occur
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Events.Then_event_listener_is_called_when_events_occur() in C:\projects\fettle\src\Tests\Core\Events.cs:line 18
8) Failed : Fettle.Tests.Core.Filtering.Projects_are_filtered.Then_only_files_matching_filter_are_candidates_for_mutation
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Filtering.Projects_are_filtered.Then_only_files_matching_filter_are_candidates_for_mutation() in C:\projects\fettle\src\Tests\Core\Filtering\Projects_are_filtered.cs:line 21
9) Failed : Fettle.Tests.Core.Filtering.Projects_are_filtered_with_multiple_filters.Then_only_files_matching_either_filter_are_candidates_for_mutation
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Filtering.Projects_are_filtered_with_multiple_filters.Then_only_files_matching_either_filter_are_candidates_for_mutation() in C:\projects\fettle\src\Tests\Core\Filtering\Projects_are_filtered_with_multiple_filters.cs:line 21
10) Failed : Fettle.Tests.Core.Filtering.Projects_are_not_filtered.Then_all_files_are_candidates_for_mutation
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Filtering.Projects_are_not_filtered.Then_all_files_are_candidates_for_mutation() in C:\projects\fettle\src\Tests\Core\Filtering\Projects_are_not_filtered.cs:line 21
11) Failed : Fettle.Tests.Core.Filtering.Source_files_are_filtered.Then_only_files_matching_filter_are_candidates_for_mutation
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Filtering.Source_files_are_filtered.Then_only_files_matching_filter_are_candidates_for_mutation() in C:\projects\fettle\src\Tests\Core\Filtering\Source_files_are_filtered.cs:line 21
12) Failed : Fettle.Tests.Core.Filtering.Source_files_are_filtered_with_multiple_filters.Then_only_files_matching_either_filter_are_candidates_for_mutation
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Filtering.Source_files_are_filtered_with_multiple_filters.Then_only_files_matching_either_filter_are_candidates_for_mutation() in C:\projects\fettle\src\Tests\Core\Filtering\Source_files_are_filtered_with_multiple_filters.cs:line 23
13) Failed : Fettle.Tests.Core.Filtering.Source_files_are_not_filtered.Then_all_files_are_candidates_for_mutation
  Expected: True
  But was:  False
   at Fettle.Tests.Core.Filtering.Source_files_are_not_filtered.Then_all_files_are_candidates_for_mutation() in C:\projects\fettle\src\Tests\Core\Filtering\Source_files_are_not_filtered.cs:line 21
14) Failed : Fettle.Tests.Core.Multiple_mutants_per_syntax_node.Then_only_one_surviving_mutant_is_returned_per_syntax_node
  Expected: 1
  But was:  0
   at Fettle.Tests.Core.Multiple_mutants_per_syntax_node.Then_only_one_surviving_mutant_is_returned_per_syntax_node() in C:\projects\fettle\src\Tests\Core\Multiple_mutants_per_syntax_node.cs:line 20
15) Error : Fettle.Tests.Core.Mutants_survive.Then_the_location_of_the_surviving_mutant_is_returned
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Fettle.Tests.Core.Mutants_survive.Then_the_location_of_the_surviving_mutant_is_returned() in C:\projects\fettle\src\Tests\Core\Mutants_survive.cs:line 25
16) Error : Fettle.Tests.Core.Mutants_survive.Then_the_relevant_post_mutation_source_code_is_returned
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Fettle.Tests.Core.Mutants_survive.Then_the_relevant_post_mutation_source_code_is_returned() in C:\projects\fettle\src\Tests\Core\Mutants_survive.cs:line 39
17) Error : Fettle.Tests.Core.Mutants_survive.Then_the_relevant_pre_mutation_source_code_is_returned
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Fettle.Tests.Core.Mutants_survive.Then_the_relevant_pre_mutation_source_code_is_returned() in C:\projects\fettle\src\Tests\Core\Mutants_survive.cs:line 33
18) Failed : Fettle.Tests.Core.Mutants_survive.Then_the_surviving_mutant_is_returned
  Expected: 1
  But was:  0
   at Fettle.Tests.Core.Mutants_survive.Then_the_surviving_mutant_is_returned() in C:\projects\fettle\src\Tests\Core\Mutants_survive.cs:line 20
```

--- a/src/Console/Console.csproj
+++ b/src/Console/Console.csproj
@@ -68,6 +68,7 @@
     </ProjectReference>
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
     <None Include="paket.references" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />

--- a/src/Console/app.config
+++ b/src/Console/app.config
@@ -1,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.2.3.0" newVersion="1.2.2.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>

--- a/src/Core/Core.csproj
+++ b/src/Core/Core.csproj
@@ -108,6 +108,7 @@
     <Compile Include="SurvivingMutant.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="app.config" />
     <None Include="paket.references" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />

--- a/src/Core/app.config
+++ b/src/Core/app.config
@@ -1,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.2.3.0" newVersion="1.2.2.0" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>

--- a/src/Tests/Tests.csproj
+++ b/src/Tests/Tests.csproj
@@ -36,9 +36,6 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="dotMemory.Unit, Version=109.0.0.0, Culture=neutral, PublicKeyToken=1010a0d8d6380325, processorArchitecture=MSIL">
-      <HintPath>..\packages\JetBrains.dotMemoryUnit.2.4.20170830.172630\lib\dotMemory.Unit.dll</HintPath>
-    </Reference>
     <Reference Include="nunit.engine">
       <HintPath>..\..\packages\NUnit.ConsoleRunner\tools\nunit.engine.dll</HintPath>
     </Reference>
@@ -129,7 +126,6 @@
     <None Include="Console\fettle.config.yml">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
-    <None Include="packages.config" />
     <None Include="paket.references" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />

--- a/src/Tests/Tests.csproj
+++ b/src/Tests/Tests.csproj
@@ -123,7 +123,9 @@
     <Compile Include="Core\No_mutants_survive.cs" />
   </ItemGroup>
   <ItemGroup>
-    <None Include="app.config" />
+    <None Include="app.config">
+      <SubType>Designer</SubType>
+    </None>
     <None Include="Console\fettle.config.yml">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>

--- a/src/Tests/app.config
+++ b/src/Tests/app.config
@@ -4,7 +4,7 @@
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
       <dependentAssembly>
         <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-1.2.2.0" newVersion="1.2.2.0" />
+        <bindingRedirect oldVersion="0.0.0.0-1.2.3.0" newVersion="1.2.2.0" />
       </dependentAssembly>
       <dependentAssembly>
         <assemblyIdentity name="System.Reflection.Metadata" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />

--- a/src/Tests/packages.config
+++ b/src/Tests/packages.config
@@ -1,4 +1,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<packages>
-  <package id="JetBrains.dotMemoryUnit" version="2.4.20170830.172630" targetFramework="net461" />
-</packages>

